### ISSUE TITLE
Stochastic Gradient Descent with L1-penalty

### ISSUE THREAD

**Issue by zarak** [rohitchauhan5/SR3#1]
![screen shot 2018-11-25 at 11 45 12 am](https://user-images.githubusercontent.com/7254237/48976344-a009c680-f0a7-11e8-989c-980db16a58fb.png)
Use the variable names as shown in the paper for consistency.

**Comment by zarak** [comment]
![f1]

[f1]: http://chart.apis.google.com/chart?cht=tx&chl=\frac{1}{2N}{(Ax-b)}^2%2B\lambda+|x|

The number of samples N is usually not indicated, but in the implementation we probably want to normalize the data by the size of the dataset to avoid having a dependence of the learning rate on N. Does that sound right?

**Comment by zarak** [comment]
Need to update [cost function](https://github.com/rohitchauhan5/SR3/blob/680ae6fab8d3043b92592f5e0737b91c4ec90bbd/R/gradient_descent_solution_3.R#L14-L16) to include penalty term.

**Comment by zarak** [comment]
Will not be implementing SGD for now.
